drv/lpc55-spi-server/src/main.rs

```diff
@@ -216,7 +216,7 @@ fn muck_with_gpios(syscon: &Syscon) {
     ];
 
     for (pin, alt, mode, slew, invert, digi, od) in
-        core::array::IntoIter::new(pin_settings)
+        IntoIterator::into_iter(pin_settings)
```

Comment:
this one was a deprecation that i fixed while I was here. the right move may be to just bump this to rust 2021 but that felt like a larger change, so I kept it at this.

Reply:
That's probably fine, though bumping to 2021 probably doesn't have _that_ many implications for this crate